lib/core_ext/active_record/connection_adapters/abstract/schema_statements.rb

```diff
@@ -1,184 +1,186 @@
-      #   :unique => true, :concurrently => true, :where => 'active')
-      # generates
-      #  CREATE UNIQUE INDEX CONCURRENTLY
-      #   index_accounts_on_branch_id_and_party_id
-      #  ON
-      #    accounts(branch_id, party_id)
-      #  WHERE
-      #    active
-      #
-      def add_index_with_concurrently(table_name, column_name, options = {})
-        creation_method = options.delete(:concurrently) ? 'CONCURRENTLY' : nil
-
-        # Whether to skip the quoting of columns. Used only for expressions like JSON indexes in which
-        # the column is difficult to target for quoting.
-        skip_column_quoting = options.delete(:skip_column_quoting) or false
-
-        index_name,
-        index_type,
-        index_columns,
-        index_options,
-        index_algorithm,
-        index_using      = add_index_options(table_name, column_name, options)
-
-        # GOTCHA:
-        #   It ensures that there is no existing index only for the case when the index
-        #   is created concurrently to avoid changing the error behavior for default
-        #   index creation.
-        #   -- zekefast 2012-09-25
-        # GOTCHA:
-        #   This check prevents invalid index creation, so after migration failed
-        #   here there is no need to go to database and clean it from invalid
-        #   indexes. But note that this handles only one of the cases when index
-        #   creation can fail!!! All other case should be procesed manually.
-        #   -- zekefast 2012-09-25
-        if creation_method.present? && index_exists?(table_name, column_name, options)
-          raise ::PgSaurus::IndexExistsError,
-                "Index #{index_name} for `#{table_name}.#{column_name}` " \
-                "column can not be created concurrently, because such index already exists."
-        end
+    module PostgreSQL # :nodoc:
+      module SchemaStatements # :nodoc:
+        # Regexp used to find the function name and function argument of a
+        # function call:
+        FUNCTIONAL_INDEX_REGEXP = /(\w+)\(((?:'.+'(?:::\w+)?, *)*)(\w+)\)/
+
+        # Regexp used to find the operator name (or operator string, e.g. "DESC NULLS LAST"):
+        OPERATOR_REGEXP = /(.+?)\s([\w\s]+)$/
+
+        # Redefine original add_index method to handle :concurrently option.
+        #
+        # Adds a new index to the table.  +column_name+ can be a single Symbol, or
+        # an Array of Symbols.
+        #
+        # ====== Creating a partial index
+        #  add_index(:accounts, [:branch_id, :party_id], :using => 'BTree'
+        #   :unique => true, :concurrently => true, :where => 'active')
+        # generates
+        #  CREATE UNIQUE INDEX CONCURRENTLY
+        #   index_accounts_on_branch_id_and_party_id
+        #  ON
+        #    accounts(branch_id, party_id)
+        #  WHERE
+        #    active
+        #
+        def add_index(table_name, column_name, options = {})
```

Comment:
No longer `_with_concurrently`? What's the rationale?

Reply:
Other methods that are losing their longer `alias_method_chain`-related names now call `super`. But this one does not.

Reply:
The updated methods were changed such that the `_without_X` were now `super`. As far as I can tell, this method completely overrides `add_index` and thus does not call `super`.